src/js/workshop-comments.js

```diff
@@ -0,0 +1,144 @@
+const commentForm = document.getElementById("comment-form");
+
+const nameDOMElement = document.getElementById("name");
+const commentDOMElement = document.getElementById("comment");
+
+commentForm.addEventListener("submit", (event) => {
+	event.preventDefault();
+
+	const nameDOMElement = document.getElementById("name");
+	const name = nameDOMElement.value.trim();
+	const commentDOMElement = document.getElementById("comment");
+	const comment = commentDOMElement.value.trim();
+
+	const submittedCommentMessage = document.querySelector(".post-failure-message");
+
+	disableFields(true);
+
+	// Hide required message
+	// Note: It's necessary to hide "required" indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment
+	document.querySelectorAll(".required").forEach(element => {
+		applyVisibility(element, false);
+	});
+	// Hide comment submission message
+	// Note: It's necessary to hide succesful/failed submitted comment message indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment
+	document.querySelectorAll(".save-response").forEach(element => {
+		applyVisibility(element, false);
+	});
+
+	// Verify input Fields
+	if ((name === "") & (comment === "")) {
+		document.querySelectorAll(".required").forEach(element => {
+			applyVisibility(element, true, "inline");
+		});
+		disableFields(false);
+		return;
+	}
+
+	if (name === "") {
+		applyVisibility(document.querySelector(".required-name"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	if (comment === "") {
+		applyVisibility(document.querySelector(".required-comment"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	const XHR = new XMLHttpRequest();
+
+	const commentForm = document.getElementById("comment-form");
+	// Bind the FormData object and the form element
+	const commentFormData = new FormData( commentForm );
+
+	// submit and add comment to top of comment section of page on successful data submission
+	XHR.addEventListener( "load", function() {
+		if (XHR.status === 200) {
+			setSubmittedComment();
+			// Clear Text Fields
+			commentForm.reset();
+		} else {
+			// Show failed comment submission message
+			submittedCommentMessage.setAttribute("style", "display: block;");
+		}
+		disableFields(false);
+	});
+
+	// Show failure message in case of error
+	XHR.addEventListener( "error", function() {
+		// Show failed comment submission message
+		submittedCommentMessage.setAttribute("style", "display: block;");
+		disableFields(false);
+	} );
+
+	// Set up our request
+	XHR.open( "POST", window.location.origin + "/api/comments" );
+
+	// The data sent is what the user provided in the form
+	const jsonFormData = JSON.stringify(Object.fromEntries(commentFormData));
```

Comment:
Did you figure out why JSON data wasn't constructed correctly last week?

Reply:
YES! It had nothing to do with `FormData( )` object or the structure of the comment form.

Going from the designers suggestions I disabled the input fields while request are being sent but all fields that are disabled are unable to be sent. This is why only the workshopID filed was being sent. (I never disabled it because it is a hidden input field)

I found a [workaround using the `readonly`](https://stackoverflow.com/questions/7357256/disabled-form-inputs-do-not-appear-in-the-request) attribute that gives a similar functionality as the `disable` attribute. Input fields that are read only are sent during form requests. I had to change the colour (to light grey) during request programatically. See the `disableFields(sendingStatus)` function.

Reply:
Nice discovery and workaround. Thanks.

Reply:
Could you add a code comment to warn the problem caused by disabling input fields and the wordaround of using `readonly`. It would also be helpful to include the url to the workaround you discovered. Thanks.